src/cpp/src/llm_pipeline.cpp

```diff
@@ -18,38 +18,12 @@
 #include "openvino/genai/lora_adapter.hpp"
 #include "lora_helper.hpp"
 #include "speculative_decoding/speculative_decoding_impl.hpp"
-#include "speculative_decoding/speculative_decoding_impl.hpp"
+#include "sampler.hpp"
+#include "lm_encoding.hpp"
 
 namespace ov {
 namespace genai {
 
-ov::genai::EncodedResults greedy_decoding(
-    ov::InferRequest& model_runner,
-    ov::Tensor prompts,
-    ov::Tensor attention_mask,
-    const GenerationConfig sampling_params,
-    const std::shared_ptr<StreamerBase> streamer,
-    std::optional<ov::Tensor> position_ids
-);
-
-ov::genai::EncodedResults multinominal_decoding(
-    ov::InferRequest& model_runner,
-    ov::Tensor prompts,
-    ov::Tensor attention_mask,
-    GenerationConfig sampling_params,
-    std::shared_ptr<StreamerBase> streamer,
-    std::optional<ov::Tensor> position_ids
-);
-
-std::pair<EncodedResults, int32_t> beam_search(
-    ov::InferRequest& lm,
-    ov::Tensor prompts,
-    ov::Tensor attention_mask,
-    GenerationConfig config,
-    std::optional<ov::Tensor> position_ids,
-    std::optional<int32_t> selected_beam_idx
-);
```

Comment:
we can also drop implementation of these functions and some other utils, which are used there